core/generated-sources/src/main/kotlin/org/jetbrains/kotlinx/dataframe/documentation/ColumnExpression.kt

```diff
@@ -0,0 +1,28 @@
+package org.jetbrains.kotlinx.dataframe.documentation
+
+import org.jetbrains.kotlinx.dataframe.api.*
+
+/**
+ * ## Column Expression
+ * In many DSLs, the lambda `expr {}` can be used to
+ * create a temporary new column by defining an expression to fill up each row.
+ *
+ * These DSLs include (but are not limited to):
+ * [The Add DSL][AddDsl.expr], [The Columns Selection DSL][ColumnsSelectionDsl.expr], and
+ * [The Create DataFrame DSL][CreateDataFrameDsl.expr].
+ *
+ * The `expr {}` call functions like a mapping statement iterating over the object it's called on.
+ */
+internal interface ColumnExpression {
+
+    /**
+     * ## Column Expression
+     * Create a temporary new column by defining an expression to fill up each row.
```

Comment:
Not sure if the word temporary fits here. In `df.select { expr { } }` it will become "permanent. I'd call it an adapter from column-based DSL to row-based (or the other way around)

Reply:
good point